test/unit/log.provider.test.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback-next-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {sinon} from '@loopback/testlab';
+import {ParsedRequest} from '@loopback/core';
+import {LogProvider} from '../..';
+
+describe('LogProvider', () => {
+  // tslint:disable-next-line:no-any
+  let spy: any;
+  beforeEach(() => {
+    spy = sinon.spy(console, 'log');
+  });
+
+  afterEach(() => {
+    spy.restore();
+  });
+
+  it('logs a value without a start time', () => {
+    const match = '/test : () => test message';
+
+    runTest(match);
+  });
+
+  it('logs a value with a start time', () => {
+    const match = 'ms: /test : () => test message';
+    const startTime = process.hrtime();
```

Comment:
You should mock `process.hrtime` and then verify the full message content including the time value. How else can you validate that the time in milliseconds is calculated correctly?

Or even better, modify `LogProvider` to make it easy to provide a custom timer function instead, while using `process.hrtime` as the default.

```ts
const startTime = [1, 0];
// mock process.hrtime() or the function used by LogFn to produce e.g. [2,2]
log(req, [], 'test message', startTime);

// assert that the message contains expected time in milliseconds (I think 1000.000002?)
// I think it would be better to actually round the value to full milliseconds and drop the fractional part
// then you can craft hrtime results to end up with `1002ms`